### ISSUE TITLE
bundle composition parameter search

### ISSUE THREAD

**Issue by owencch** [IBM/FHIR#2689]
Why can't the information be obtained in the second picture(Bundle?composition:Composition.status=final)?
I tried both v4.7.1 and v4.8.3 and the results are the same (Total always is "0")
![20210816_B](https://user-images.githubusercontent.com/16168209/129648806-ed990662-f851-4bdc-8c36-f25173784919.png)
![20210816_A](https://user-images.githubusercontent.com/16168209/129648948-0cc3d71b-8175-42aa-a55b-9840e5f724b6.png)

**Comment by lmsurpre** [comment]
Seems like a specific instance of the more general case we'd need to support for #2732

**Comment by lmsurpre** [comment]
Note:  Bundle also has a very similar search parameter Bundle-message.  Its just like Bundle-composition but for messages instead of documents.

**Comment by lmsurpre** [comment]
@owencch given the cost of storing all contained resources in all cases, we thinking it might be better for you to define your own custom search parameters for just the fields you want.  For example, in this case you could introduce your own search parameter for Bundle-composition-status with an expressions like `Bundle.where(type='document').entry[0].resource.as(Composition).status`.  Let us know if you need some hints/examples for how to do that.

**Comment by owencch** [comment]
Yes, We need some hints/examples to illustrate how to do this, please let
us know.

Lee Surprenant ***@***.***> 於 2021年8月30日 週一 下午11:00寫道：

**Comment by lmsurpre** [comment]
@owencch the docs for that are at https://ibm.github.io/FHIR/guides/FHIRSearchConfiguration#11-tenant-specific-parameters.  assuming your are using a single tenant named "default", you'd populate config/default/extension-search-parameters.json with a collection that has one or more custom search parameter definition.

There is an example in 1.1.1 of that guide and you can try modifying that into what you want.  Something like this:
```
{
   "resourceType": "Bundle",
   "type": "collection",
   "entry": [{
      "fullUrl": "http://ibm.com/fhir/SearchParameter/Bundle-composition-status",
      "resource": {
         "resourceType": "SearchParameter",
         "id": "Bundle-composition-status",
         "url": "http://ibm.com/fhir/SearchParameter/Bundle-composition-status",
         "version": "4.0.1",
         "name": "composition-status",
         "status": "draft",
         "experimental": true,
         "description": "The status of the Composition in this document bundle",
         "code": "composition-status",
         "base": ["Bundle"],
         "type": "token",
         "expression": "Bundle.where(type='document').entry[0].resource.as(Composition).status",
         "multipleOr": true,
         "multipleAnd": true,
         "modifier": []
      }
   }
}
```

**Comment by michaelwschroeder** [comment]
Adding some design notes
[issue2689DesignNotes.docx](https://github.com/IBM/FHIR/files/7695309/issue2689DesignNotes.docx)
